src/main/java/org/gridsuite/study/server/repository/StudyEntity.java

```diff
@@ -6,36 +6,83 @@
  */
 package org.gridsuite.study.server.repository;
 
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+import lombok.*;
 import org.gridsuite.study.server.dto.LoadFlowStatus;
 
-import java.util.UUID;
+import javax.persistence.*;
 
 /**
  * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ * @author Chamseddine Benhamed <chamseddine.benhamed at rte-france.com>
  */
-public interface StudyEntity extends BasicStudyEntity {
 
-    void setStudyName(String studyName);
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Entity
+@Builder
+@Table(name = "study")
+public class StudyEntity implements BasicStudyEntity {
+
+    @Id
+    @GeneratedValue(strategy  =  GenerationType.AUTO)
+    @Column(name = "id")
+    private UUID id;
+
+    @Column(name = "userId", nullable = false)
+    private String userId;
+
+    @Column(name = "studyName", nullable = false)
+    private String studyName;
+
+    @Column(name = "creationDate")
+    private LocalDateTime date;
 
-    UUID getNetworkUuid();
+    @Column(name = "networkUuid")
+    private UUID networkUuid;
 
-    String getNetworkId();
+    @Column(name = "networkId")
+    private String networkId;
 
-    String getDescription();
+    @Column(name = "description")
+    private String description;
 
-    String getCaseFormat();
+    @Column(name = "caseFormat")
+    private String caseFormat;
 
-    UUID getCaseUuid();
+    @Column(name = "caseUuid")
+    private UUID caseUuid;
 
-    boolean isCasePrivate();
+    @Column(name = "casePrivate")
+    private boolean casePrivate;
 
-    boolean isPrivate();
+    @Column(name = "isPrivate")
+    private boolean isPrivate;
 
-    LoadFlowStatus getLoadFlowStatus();
+    @Column(name = "loadFlowStatus")
+    @Enumerated(EnumType.STRING)
+    private LoadFlowStatus loadFlowStatus;
 
-    LoadFlowResultEntity getLoadFlowResult();
```

Comment:
In cassandra, the loadflow result and the loadflow parameters were kind of embedded in the study. Can we do the same here and not have entities with ids for these ?

Reply:
We can look into this, but if doing entities for everything is the recommended approach for jpa, then we will do that. Some articles to look into:
https://www.baeldung.com/jpa-mapping-single-entity-to-multiple-tables
https://vladmihalcea.com/the-best-way-to-map-a-onetoone-relationship-with-jpa-and-hibernate/
https://stackoverflow.com/questions/43599840/mapping-an-embeddable-in-a-separate-table

Not sure for @embeddable since you can't to do this transitive embeddables :(